core/Tracker/Request.php

```diff
@@ -634,6 +635,14 @@ public function getUserAgent()
         return Common::getRequestVar('ua', $default, 'string', $this->params);
     }
 
+    public function getClientHints()
+    {
+        // use headers as default if no data was send with the tracking request
+        $default = Http::getClientHintsFromServerVariables();
+
+        return Common::getRequestVar('uadata', $default, 'json', $this->params);
```

Comment:
This new parameter would need to be documented on https://developer.matomo.org/api-reference/tracking-api, right?